src/libs/localStore/sharedMemory.ts

```diff
@@ -0,0 +1,51 @@
+/*
+  This shared memory is meant to store generic data accessible by
+  Cozy webviews. It is a very simple implementation for the moment.
+*/
+type SharedMemorySlug = 'mespapiers'
+type SharedMemoryKey = 'scanDocument'
+type SharedMemoryData = string | undefined
+
+const sharedMemory = {
+  mespapiers: {
+    scanDocument: undefined as SharedMemoryData
+  }
+}
+
+export const getSharedMemory = (
+  slug: SharedMemorySlug,
+  key: SharedMemoryKey
+): SharedMemoryData => {
+  return sharedMemory[slug][key]
+}
+
+export const storeSharedMemory = (
+  slug: SharedMemorySlug,
+  key: SharedMemoryKey,
+  value: SharedMemoryData
+): void => {
+  sharedMemory[slug][key] = value
+}
+
+export const removeSharedMemory = (
+  slug: SharedMemorySlug,
+  key: SharedMemoryKey
+): void => {
+  sharedMemory[slug][key] = undefined
+}
+
+export const sharedMemoryLocalMethods = {
+  getSharedMemory: (
+    slug: SharedMemorySlug,
```

Comment:
@acezard do you know if we can add a mechanism to inject the `slug` automatically from the nativeIntent code? This would prevent a cozy-app to inject the wrong slug and steal data from other cozy-app.

Reply:
yes but we would need to update every cozy-app if we add this in cozy-intent. or we could also guess the slug from react-native which requires only updating the mobile app